From a941875fe4cc3effd0f79c453a5fabc04001a3ec Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Thu, 16 Oct 2014 22:17:54 +0800
Subject: [PATCH] MLK-9617-17 usb: common: otg: clear b_bus_req in
 b_peripheral state

This patch is clearing b_bus_req input to be 0 when entering b_peripheral state.
b_bus_req is only TRUE when b device wants to use the bus and switch to be host
role.

Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit eb610e73aa38bbb45aed6a79265cd79fe80cdd05)
---
 drivers/usb/common/usb-otg-fsm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/common/usb-otg-fsm.c b/drivers/usb/common/usb-otg-fsm.c
index 72f1280..43416c2 100644
--- a/drivers/usb/common/usb-otg-fsm.c
+++ b/drivers/usb/common/usb-otg-fsm.c
@@ -166,6 +166,7 @@ static int otg_set_state(struct otg_fsm *fsm, enum usb_otg_state new_state)
 		otg_loc_sof(fsm, 0);
 		otg_set_protocol(fsm, PROTO_GADGET);
 		otg_loc_conn(fsm, 1);
+		fsm->b_bus_req = 0;
 		break;
 	case OTG_STATE_B_WAIT_ACON:
 		otg_chrg_vbus(fsm, 0);
-- 
1.8.0

